webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/RuleValService.scala

```diff
@@ -90,11 +104,11 @@ class RuleValServiceImpl(
    * We must exclude variable from ncf technique, that are processed differently, because
    * the provided value is not managed by rudder (it is on a cfengine file elsewhere).
    */
-  override def lookupNodeParameterization(variables: Seq[Variable]): InterpolationContext => IOResult[Map[String, Variable]] = {
+  override def lookupNodeParameterization(variables: Map[(List[String],String),Variable]): InterpolationContext => IOResult[Map[(List[String],String) , Variable]] = {
```

Comment:
I think that `(List[String], String)` really need a dedicated data structure like `ComponentId`